src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -724,6 +690,37 @@ public static void onWorldLoad(@NotNull final World world)
         }
     }
 
+    public static boolean backupColonyData()
+    {
+        if (numWorldsLoaded > 0 && saveNeeded)
+        {
+            saveColonies();
+        }
+
+        @NotNull final File file = getSaveLocation();
+        @NotNull final File targetFile = getBackupSaveLocation(new Date());
+        if (!file.exists())
+        {
+            return true;
+        }
+        else if (targetFile.exists())
+        {
+            return false;
+        }
+
+        try
+        {
+            Files.copy(file.toPath(), targetFile.toPath());
+        }
+        catch (IOException e)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)